vernac/vernacentries.ml

```diff
@@ -607,6 +607,24 @@ let vernac_assumption ~atts discharge kind l nl =
   let status = ComAssumption.do_assumptions ~poly:atts.polymorphic ~program_mode:atts.program ~scope ~kind nl l in
   if not status then Feedback.feedback Feedback.AddedAxiom
 
+let set_template_check b =
+ let typing_flags = Environ.typing_flags (Global.env ()) in
+ Global.set_typing_flags { typing_flags with Declarations.check_template = b }
+
+let is_template_check () =
+  let typing_flags = Environ.typing_flags (Global.env ()) in
+  typing_flags.Declarations.check_template
+
+let () =
+  let tccheck =
+    { optdepr = true;
+      optname = "Template universe check";
+      optkey = ["Template"; "Polymorphic"; "Inductive"; "Check"];
+      optread = (fun () -> is_template_check ());
```

Comment:
What do you guys prefer? The longer name is more explicit... but maybe overkill as we don't expect many users and this is already deprecated?

Reply:
I prefer the shorter name.